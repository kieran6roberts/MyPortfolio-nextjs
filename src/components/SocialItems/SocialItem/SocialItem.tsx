interface SOCIAL_ITEM {
    children: React.ReactElement;
    label: string;
    path: string;
}

export default function SocialItem({ path, label, children}: SOCIAL_ITEM): React.ReactElement {
    return (
        <li className="mx-2 mb-2">
            <a aria-label={label}
            className="text-md focus:outline-none focus:ring-4 focus:ring-yellow-400"
            href={path}
            target="_blank">
                {children}
            </a>
        </li>
    )
};