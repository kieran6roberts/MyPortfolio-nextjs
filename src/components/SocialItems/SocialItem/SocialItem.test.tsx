import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import SocialItem from "@/components/SocialItems/SocialItem/SocialItem";

let documentBody: RenderResult;

describe("<SocialItems {...props> />", () => {
    const props = {
        path: "/social",
        label: "to my social page",
        children: <h2>child</h2>
    };

    test("renders", () => {
        documentBody = render(<SocialItem {...props}/>);
        
        expect(documentBody.getByRole("link", { name: /to my social page/})).toBeInTheDocument();
    });
    
    test("props are applied", () => {
        documentBody = render(<SocialItem {...props}/>);
        
        expect(documentBody.getByRole("link", { name: /to my social page/})).toHaveAttribute("aria-label", "to my social page");
        expect(documentBody.getByRole("link", { name: /to my social page/})).toHaveAttribute("href", "/social");
        expect(documentBody.getByRole("link", { name: /to my social page/})).toContainElement(documentBody.getByRole("heading"));
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<SocialItem {...props} />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});