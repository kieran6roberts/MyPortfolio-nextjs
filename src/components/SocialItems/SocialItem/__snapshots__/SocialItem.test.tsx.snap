// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialItems {...props> /> render snapshot 1`] = `
<li
  className="mx-3 mb-2 2xl:mx-8"
>
  <a
    aria-label="to my social page"
    className="text-md"
    href="/social"
    rel="noopener"
    target="_blank"
  >
    <h2>
      child
    </h2>
  </a>
</li>
`;
